sympy/printing/conventions.py

```diff
@@ -7,7 +7,7 @@
 from collections.abc import Iterable
 from sympy import Derivative
 
-_name_with_digits_p = re.compile(r'^([a-zA-Z]+)([0-9]+)$')
+_name_with_digits_p = re.compile(r'^([^\W\d_]+)([0-9]+)$', re.U)
```

Comment:
Why isn't this just `\w`?

Reply:
Match `"w11"` against `(\w+)([0-9]+)` gives `"w1"` and `"1"` because the first bit tries to match as much as possible. We could instead use `\w+?` for the first capturing group to match as few characters as possible, but I think it's a bit clearer to say: "not(not Unicode word character or digit or underscore)", which is the same as "Unicode word character and not digit and not underscore". I am going to add a test to cover `"w11"` as well.

Sorry about omitting the release notes; I thought this was a small enough change that didn't require any such notes. Will keep this in mind for the future!

Reply:
Looking at what you've got, I think we should also be consistent with `[0-9]` vs. `\d`. I think `\d` is better as that matches Unicode digits (https://www.fileformat.info/info/unicode/category/Nd/list.htm). That includes digits in other scripts than Latin, which you would presumably want to handle the same way with respect to subscripting.

Reply:
Or even something like `Symbol('x𝟘')`.

Reply:
You are right! Didn't think about the possibility of having non-Latin digits, so I've just fixed the regex and added a few more tests.